### PR TITLE
Feeless Ante Test

### PR DESCRIPTION
**Pull request by Kissadada**
## Implementation details

add 'ante_test.go' for testing freeless report ante
 
## Please ensure the following requirements are met before submitting a pull request:

- [ ] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: x/oracle/ante/ante_test.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+	msgs := []sdk.Msg{types.NewMsgReportData(suite.requestId, []types.RawReport{}, testapp.Validators[0].ValAddress)}
+	stubTx := &MyStubTx{Msgs: msgs}
+
+	suite.mockAnte.On("Ante", suite.ctx.WithMinGasPrices(sdk.DecCoins{}), stubTx, false)
+	ctx, err := suite.feelessAnte(suite.ctx, stubTx, false)
+
+	suite.mockAnte.AssertExpectations(suite.T())
+	suite.Require().Equal(ctx.MinGasPrices(), suite.ctx.MinGasPrices())
+	suite.Require().Equal(err, nil)
```
Comment (songwongtp) [inline x/oracle/ante/ante_test.go:74]: `suite.Require().NoEqual(err)`

File: x/oracle/ante/ante_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+	ctx, err := suite.feelessAnte(suite.ctx, stubTx, false)
+
+	suite.Require().Equal(ctx, suite.ctx)
+	suite.Require().Error(err)
+}
+
+func (suite *AnteTestSuit) TestValidReport() {
+	reportMsgs := []sdk.Msg{types.NewMsgReportData(suite.requestId, []types.RawReport{}, testapp.Validators[0].ValAddress)}
+	autzMsg := authz.NewMsgExec(testapp.Alice.Address, reportMsgs)
```
Comment (songwongtp) [inline x/oracle/ante/ante_test.go:89]: `authzMsg` ?

File: x/oracle/ante/ante_test.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	ctx, err := suite.feelessAnte(suite.ctx, stubTx, false)
+
+	suite.mockAnte.AssertExpectations(suite.T())
+	suite.Require().Equal(ctx.MinGasPrices(), suite.ctx.MinGasPrices())
+	suite.Require().Equal(err, nil)
+}
+
+func (suite *AnteTestSuit) TestNoAuthzReport() {
+	reportMsgs := []sdk.Msg{types.NewMsgReportData(suite.requestId, []types.RawReport{}, testapp.Validators[0].ValAddress)}
+	autzMsg := authz.NewMsgExec(testapp.Bob.Address, reportMsgs)
```
Comment (songwongtp) [inline x/oracle/ante/ante_test.go:102]: ditto

File: x/oracle/ante/ante_test.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+	stubTx := &MyStubTx{Msgs: []sdk.Msg{&autzMsg}}
+
+	_, err := suite.feelessAnte(suite.ctx, stubTx, false)
+
+	suite.mockAnte.AssertNumberOfCalls(suite.T(), "Ante", 0)
+	suite.Require().Error(err)
+}
+
+func (suite *AnteTestSuit) TestNotReportMsg() {
+	requetMsg := types.NewMsgRequestData(1, BasicCalldata, 1, 1, BasicClientID, testapp.Coins100000000uband, testapp.TestDefaultPrepareGas, testapp.TestDefaultExecuteGas, testapp.FeePayer.Address)
```
Comment (songwongtp) [inline x/oracle/ante/ante_test.go:123]: requestMsg

File: x/oracle/ante/ante_test.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+	ctx, err := suite.feelessAnte(suite.ctx, stubTx, false)
+
+	suite.mockAnte.AssertExpectations(suite.T())
+	suite.Require().Equal(ctx, suite.ctx)
+	suite.Require().Equal(err, nil)
+}
+
+func (suite *AnteTestSuit) TestNotReportMsgButReportOnlyBlock() {
+	suite.ctx = suite.ctx.WithBlockHeight(0)
+	requetMsg := types.NewMsgRequestData(1, BasicCalldata, 1, 1, BasicClientID, testapp.Coins100000000uband, testapp.TestDefaultPrepareGas, testapp.TestDefaultExecuteGas, testapp.FeePayer.Address)
```
Comment (songwongtp) [inline x/oracle/ante/ante_test.go:136]: ditto

File: x/oracle/ante/ante_test.go [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+	reportMsgs := []sdk.Msg{types.NewMsgReportData(suite.requestId, []types.RawReport{}, testapp.Validators[0].ValAddress)}
+	autzMsg := authz.NewMsgExec(testapp.Alice.Address, reportMsgs)
+	stubTx := &MyStubTx{Msgs: []sdk.Msg{&autzMsg}}
+
+	suite.mockAnte.On("Ante", suite.ctx.WithMinGasPrices(sdk.DecCoins{}), stubTx, false)
+	ctx, err := suite.feelessAnte(suite.ctx, stubTx, false)
+
+	suite.mockAnte.AssertExpectations(suite.T())
+	suite.Require().Equal(ctx.MinGasPrices(), suite.ctx.MinGasPrices())
+	suite.Require().Equal(err, nil)
```
Comment (songwongtp) [inline x/oracle/ante/ante_test.go:97]: ditto

File: x/oracle/ante/ante_test.go [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+	_, err := suite.feelessAnte(suite.ctx, stubTx, false)
+
+	suite.mockAnte.AssertNumberOfCalls(suite.T(), "Ante", 0)
+	suite.Require().EqualError(err, sdkerrors.ErrUnauthorized.Wrap("authorization not found").Error())
+}
+
+func (suite *AnteTestSuit) TestNotValidReport() {
+	reportMsgs := []sdk.Msg{types.NewMsgReportData(suite.requestId+1, []types.RawReport{}, testapp.Validators[0].ValAddress)}
+	autzMsg := authz.NewMsgExec(testapp.Alice.Address, reportMsgs)
```
Comment (songwongtp) [inline x/oracle/ante/ante_test.go:113]: ditto

**Review by songwongtp — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**